ci/rt.sh

```diff
@@ -0,0 +1,254 @@
+# This script is desined for UPP regression tests run by UPP developer.
+# Wen Meng, 12/2020, First version.
+# Fernando Andrade-Maldonado 5/2023 rework for CLI Options
+# Fernando Andrade-Maldonado / Wen Meng 9/2023 Add Hercules, fix typos, and refactor
+#
+######################################################################
+
+git_branch="develop"
+git_url="https://github.com/NOAA-EMC/UPP.git"
+clone_on="no"
+
+while getopts a:w:h:r:t:b:u:c opt; do
+  case $opt in
+    a) accnr=${OPTARG}
+        ;;
+    w) workdir=${OPTARG}
+        ;;
+    h) homedir=${OPTARG}
+        ;;
+    r) rundir=${OPTARG}
+        ;;
+    t) test_v=${OPTARG}
+        ;;
+    b) git_branch=${OPTARG}
+        ;;
+    u) git_url=${OPTARG}
+        ;;
+    c) clone_on="yes"
+	;;
+  esac
+done
+
+#UPP working copy
+test_v=${test_v:-`pwd`/../}
+if [[ $clone_on == "yes" ]]; then
+  rm -rf $test_v
+  mkdir -p $test_v
+  git clone -b $git_branch $git_url $test_v
+fi
+export svndir=${test_v}
+
+#Assume a nems account to run with
+accnr=${accnr:-"nems"}
+
+#Build UPP executable
+build_exe=yes
+
+#Choose run specific model
+run_nmmb=yes
+run_gfs=yes
+run_gefs=yes
+run_fv3r=yes
+run_rap=yes
+run_hrrr=yes
+run_hafs=yes
+run_rtma=yes
+
+#find machine
+mac=$(hostname | cut -c1-1)
+mac2=$(hostname | cut -c1-2)
+mac3=$(hostname | cut -c1-4)
+if [ $mac = v -o $mac = m  ] ; then   # WCOSS_D (venus and mars)
+ export machine=WCOSS_DELL_P3
+ export homedir=${homedir:-"/u/Wen.Meng/noscrubd/ncep_post/post_regression_test_new"}
+ export rundir=${rundir:-"/gpfs/dell2/ptmp/$USER"}
+elif [ $mac2 = hf ]; then #for HERA
+ export machine=HERA
+ export homedir=${homedir:-"/scratch2/NAGAPE/epic/UPP/test_suite"}
+ export rundir=${rundir:-"/scratch1/NCEPDEV/stmp2/${USER}"}
+ module use /scratch1/NCEPDEV/nems/role.epic/hpc-stack/libs/intel-2022.1.2/modulefiles/stack
+ module load hpc/1.2.0
+ module load hpc-intel/2022.1.2
+ module load hpc-impi/2022.1.2
+ module load prod_util
+elif [ $mac = O ] ; then
+ export machine=ORION
+ export homedir=${homedir:-"/work/noaa/epic/UPP"}
+ export rundir=${rundir:-"/work2/noaa/stmp/$USER"}
+ module use /work/noaa/epic-ps/role-epic-ps/hpc-stack/libs/intel-2022.1.2/modulefiles/stack
+ module load hpc/1.2.0
+ module load hpc-intel/2022.1.2
+ module load hpc-impi/2022.1.2
+ module load prod_util/1.2.2
+elif [ $mac3 = herc ] ; then
+ export machine=HERCULES
+ export homedir=${homedir:-"/work/noaa/epic/UPP"}
+ export rundir=${rundir:-"/work2/noaa/stmp/$USER"}
+ module use /work/noaa/epic/role-epic/spack-stack/hercules/spack-stack-dev-20230717/envs/unified-env/install/modulefiles/Core
+ module load stack-intel/2021.9.0
+ module load stack-intel-oneapi-mpi/2021.9.0
+ module load jasper/2.0.32
+ module load prod-util/1.2.2
+fi
+
+#set working directory
+export workdir=${workdir:-"`pwd`/work-upp-${machine}"}
+rm -rf $workdir
+mkdir -p $workdir
+
+#differentiates for orion and hercules
+export rundir="${rundir}-upp-${machine}"
```

Comment:
@FernandoAndrade-NOAA Do you think using "${rundir}/upp-machine" would make it easier for users to clean the temporary space periodically on R&D platforms if they run the UPP RTs by themselves? For example,  /scratch1/NCEPDEV/stmp2/Wen.Meng-upp-HERA  VS  /scratch1/NCEPDEV/stmp2/Wen.Meng/upp-HERA

Reply:
That may be a little better organized yes. I can go ahead and update that. The main purpose of this was just to avoid conflicts in the the shared filesystems. Would you have a preference for a different directory name or should /upp-${machine} be fine?

Reply:
@FernandoAndrade-NOAA Your update looks good for me.